modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -58,32 +58,37 @@ namespace gimpl {
     struct Data;
     struct RcDesc;
 
-    struct GAPI_EXPORTS RMatMediaAdapterBGR final: public cv::RMat::Adapter
+    struct GAPI_EXPORTS RMatMediaFrameAdapter final: public cv::RMat::Adapter
     {
-        explicit RMatMediaAdapterBGR(const cv::MediaFrame& frame) : m_frame(frame) { };
+        explicit RMatMediaFrameAdapter(const cv::MediaFrame& frame,
+            std::function<cv::Mat(const GFrameDesc&, const cv::MediaFrame::View&)> frameViewToMat) :
+            m_frame(frame),
+            m_frameViewToMat(frameViewToMat)
+        {
+            const auto& d = m_frame.desc();
+            const auto& fv = m_frame.access(cv::MediaFrame::Access::R);
+            m_matDesc = cv::descr_of(m_frameViewToMat(d, fv));
+        };
 
         virtual cv::RMat::View access(cv::RMat::Access a) override
         {
-            auto view = m_frame.access(a == cv::RMat::Access::W ? cv::MediaFrame::Access::W
-                                                                : cv::MediaFrame::Access::R);
-            auto ptr = reinterpret_cast<uchar*>(view.ptr[0]);
-            auto stride = view.stride[0];
+            auto fv = m_frame.access(a == cv::RMat::Access::R ? cv::MediaFrame::Access::R
+                                                              : cv::MediaFrame::Access::W);
```

Comment:
There's no guarantee in only `R` and `W` being used exclusively, there may be `RW` introduced one day.

This code will break then.

Reply:
Added mapping lamda, but it throws exception if `cv::RMat::Access` is not `R` or `W` for now.